code/modules/mob/mob.dm

```diff
@@ -1356,6 +1356,10 @@ GLOBAL_LIST_INIT(slot_equipment_priority, list( \
 	SEND_SIGNAL(src, COMSIG_MOB_UPDATE_SIGHT)
 	sync_lighting_plane_alpha()
 
+/mob/proc/set_sight(datum/vision_override/O)
+	vision_type = O
```

Comment:
Can you check if you can `qdel` the old `vision_type`?
You will have to make sure that every time vision_type is set that a new type is being used there. And not a variable that is saved on the item or such. Else you remove the items vision_type.